lib/src/repository/check_repository.dart

```diff
@@ -58,47 +64,86 @@ Future<Repository?> checkRepository(PackageContext context) async {
         failVerification('Repository has no `pubspec.yaml`.');
         return result();
       }
-      final pubspecPath = p.join(url.path, 'pubspec.yaml');
-      if (!pubspecFiles.contains(pubspecPath)) {
-        // TODO: detect if any of them is matching
-        failVerification(
-            'Repository has no `pubspec.yaml` in location: `${url.path}/`.');
-        return result();
+
+      Future<_PubspecMatch> matchRepoPubspecYaml(String path) async {
+        // checkout the pubspec.yaml at the assumed location
+        final content = await repo.showStringContent(
+          branch!,
+          path,
+          maxOutputBytes: _maxPubspecBytes,
+        );
+        final gitPubspec = Pubspec.parseYaml(content);
+        // verification steps
+        if (gitPubspec.name != context.pubspec.name) {
+          return _PubspecMatch(
+              path,
+              false,
+              'Repository `$path` name missmatch: '
+              '`${gitPubspec.name}` != `${context.pubspec.name}`.');
+        }
+        final gitRepoOrHomepage = gitPubspec.repositoryOrHomepage;
+        if (gitRepoOrHomepage == null) {
+          return _PubspecMatch(path, true,
+              'Repository `$path` has no `repository` or `homepage` URL.');
+        }
+        final gitRepoUrl = RepositoryUrl.tryParse(gitRepoOrHomepage);
+        if (gitRepoUrl?.baseUrl != url.baseUrl) {
+          return _PubspecMatch(path, true,
+              'Repository `$path` URL missmatch: expected `${url.baseUrl}` but got `${gitRepoUrl?.baseUrl}`.');
+        }
+        if (gitPubspec.version == null) {
+          return _PubspecMatch(
+              path, true, 'Repository `$path` has no version.');
+        }
+        if (gitPubspec.toJson().containsKey('publish_to ')) {
```

Comment:
Whitespace is probably an issue here :D
```suggestion
        if (gitPubspec.toJson().containsKey('publish_to')) {
```